controllers/istio/istio_controller.go

```diff
@@ -91,7 +91,10 @@ func (r *IstioReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 
 	log.Info("Reconciliation done. Updating status.")
 	err = r.updateStatus(ctx, &istio, err)
-
+	if errors.IsConflict(err) {
+		log.Info("Status update failed. Requeuing reconciliation")
+		return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+	}
```

Comment:
BTW I don't think we need to check for conflicts here, since we use PATCH to update the status.

Reply:
Or, instead of removing, I think moving the conflict check higher up might be a better idea, as it would take care of all possible conflicts in one single place (with no code duplication). I think we have something like this in 2.x.